### DISCUSSION TITLE
SMB connections

### DISCUSSION THREAD

**Discussion by firemen01** [DelphiWorlds/Kastri#246] [General]
There was a need to implement support for connecting from an Android application to a shared resource in Windows. I was surprised to find that this is far from a trivial task and I did not find any working solutions anywhere. There is one library here that the author abandoned (https://github.com/LongDirtyAnimAlf/Delphi-Android-SMB/) 
Users write that everything works on Delphi XE8, but on newer ones an initialization error pops up. Tell me what changes have occurred in Delphi, maybe there is a way to somehow run it on Delphi 12. A very useful thing when you need to have access to large amounts of data stored on a PC.

**Comment by DelphiWorlds** [comment]
I ran the app in that repo and used [Device Lens]() to check the callstack for when the error occurs, which reveals this:

<img width="819" alt="image" src="https://github.com/user-attachments/assets/4ef67fe4-00c2-4d94-81be-841f942fd9df">

I have highlighted the cause, which is that the networking call is being performed in the main thread, and Android is objecting to it. Not sure how the app could work in XE8, since it's an Android restriction, not something to do with Delphi. The answer would likely be to execute the relevant calls inside a thread, and synchronize to the main thread with the results
